front/src/applications/editor/tools/rangeEdition/components/RangeEditionLeftPanel.tsx

```diff
@@ -53,6 +72,66 @@ const RangeEditionLeftPanel = () => {
     });
   };
 
+  const unselectSwitch = (sw: string) => () => {
+    setState({
+      selectedSwitchs: selectedSwitchs.filter((s) => s !== sw),
+    });
+  };
+
+  const searchRoutesFromSwitch = async () => {
+    const body = selectedSwitchs.reduce(
+      (accumulator, value) => ({ ...accumulator, [value]: null }),
+      {}
+    );
+    setState({
+      optionsState: { type: 'loading' },
+    });
+    const routesCandidate = await getRoutesFromSwitch({
+      infraId: infraID as number,
+      body,
+    }).unwrap();
+    setSwitchsRoutesCandidate(routesCandidate as string[]);
+    setState({
+      optionsState: { type: 'idle' },
+    });
+  };
+
+  const getTrackRanges = async (routes: string) => {
+    const trackRangesResults = await getTrackRangesByRoutes({
+      infraId: infraID as number,
+      routes,
+    }).unwrap();
+    if (trackRangesResults[0].type === 'Computed') {
+      const trackRanges = trackRangesResults[0].track_ranges.map((trackRange) => {
+        const { begin, end, track } = trackRange;
+        return {
+          begin,
+          end,
+          track,
+          applicable_directions: trackRange.direction,
+        };
+      });
+      const newEntity = cloneDeep(entity) as SpeedSectionEntity;
+      newEntity.properties.track_ranges = trackRanges;
+      setState({
+        optionsState: { type: 'idle' },
+      });
+      setState({
+        entity: newEntity,
+      });
+    }
+  };
+
+  useEffect(() => {
+    if (switchsRoutesCandidate.length > 0) {
+      getTrackRanges(switchsRoutesCandidate[0]);
+    }
+  }, [switchsRoutesCandidate]);
+
+  useEffect(() => {
+    if (!selectedSwitchs.length) setSwitchsRoutesCandidate([]);
+  }, [selectedSwitchs]);
```

Comment:
I think each time we add or remove a switch, we should reset the list of the routeCandidates
```suggestion
  useEffect(() => {
    if (switchsRoutesCandidate.length) setSwitchsRoutesCandidate([]);
  }, [selectedSwitchs]);
```

Reply:
done